payu/models/um.py

```diff
@@ -141,13 +140,18 @@ def setup(self):
                                            work_path=self.work_path)
         os.environ.update(um_vars)
 
-        # The above needs to be done in parexe also.
-        # FIXME: a better way to do this or remove.
-        parexe = os.path.join(self.work_path, 'parexe')
-        for line in fileinput.input(parexe, inplace=True):
-            line = line.format(input_path=self.input_paths[0],
-                               work_path=self.work_path)
-            print(line, end='')
+        # parexe removed from newer configurations - retain the
+        # old processing if file exists for backwards compatibility
+        try:
+            # The above needs to be done in parexe also.
+            # FIXME: a better way to do this or remove.
+            parexe = os.path.join(self.work_path, 'parexe')
+            for line in fileinput.input(parexe, inplace=True):
+                line = line.format(input_path=self.input_paths[0],
+                                   work_path=self.work_path)
+                print(line, end='')
+        except FileNotFoundError:
+            pass
```

Comment:
Alternatively `os.path.isfile()` could be used to see if parexe file exists, instead of the try except block - but both work fine
```suggestion
        # The above needs to be done in parexe also, if it exists
        parexe = os.path.join(self.work_path, 'parexe')
        if os.path.isfile(parexe):
            # FIXME: a better way to do this or remove.
            for line in fileinput.input(parexe, inplace=True):
                line = line.format(input_path=self.input_paths[0],
                                   work_path=self.work_path)
                print(line, end='')
```

Reply:
Great idea, I think that makes it clearer to read!